packages/@ember/-internals/glimmer/lib/renderer.ts

```diff
@@ -290,14 +291,15 @@ export class Renderer {
 
   readonly _runtimeResolver: ResolverImpl;
 
-  static create(props: {
-    document: SimpleDocument;
-    env: { isInteractive: boolean; hasDOM: boolean };
-    rootTemplate: TemplateFactory;
-    _viewRegistry: any;
-    builder: any;
-  }): Renderer {
-    let { document, env, rootTemplate, _viewRegistry, builder } = props;
+  static create(props: { _viewRegistry: any }): Renderer {
+    let { _viewRegistry } = props;
+    let document = getOwner(props).lookup('service:-document') as SimpleDocument;
+    let env = getOwner(props).lookup('-environment:main') as {
+      isInteractive: boolean;
+      hasDOM: boolean;
+    };
+    let rootTemplate = getOwner(props).lookup(P`template:-root`) as TemplateFactory;
+    let builder = getOwner(props).lookup('service:-dom-builder') as IBuilder;
```

Comment:
```suggestion
    let owner = getOwner(props);
    let rootTemplate = owner.lookup(P`template:-root`) as TemplateFactory;
    let builder = owner.lookup('service:-dom-builder') as IBuilder;
```

Better to do the lookup once rather than multiple times